derp/derp_server.go

```diff
@@ -57,11 +60,18 @@ type Server struct {
 
 	mu          sync.Mutex
 	closed      bool
-	netConns    map[net.Conn]chan struct{} // chan is closed when conn closes
+	netConns    map[Conn]chan struct{} // chan is closed when conn closes
 	clients     map[key.Public]*sclient
 	clientsEver map[key.Public]bool // never deleted from, for stats; fine for now
 }
 
+type Conn interface {
```

Comment:
This is a weird type, especially without docs. (No way to Write to it??, the reader asks.)